test/snippet/core/algorithm/configuration.cpp

```diff
@@ -1,88 +1,96 @@
+#include <iostream>
+
+//! [configuration_setup]
 #include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
 
 using namespace seqan3;
 
-struct bar
+enum struct my_id : int
 {
-    float value = 0;
+    bar_id,
+    foo_id
 };
 
-struct foo
+struct bar : public pipeable_config_element
 {
-    int value = 0;
+    static constexpr my_id id{my_id::bar_id};
+
+    bar()                        = default;
+    bar(bar const &)             = default;
+    bar(bar &&)                  = default;
+    bar & operator=(bar const &) = default;
+    bar & operator=(bar &&)      = default;
+    ~bar()                       = default;
+
+    bar(float v) : value(v)
+    {}
+
+    float value;
 };
 
-struct with_foo_adaptor : public detail::configuration_fn_base<with_foo_adaptor>
+template <typename t>
+struct foo : public pipeable_config_element
 {
-    template <typename configuration_t>
-    constexpr auto invoke(configuration_t && cfg,
-                          int new_v) const
-        requires detail::is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
-    {
-        return std::forward<configuration_t>(cfg).push_front(foo{new_v});
-    }
+    static constexpr my_id id{my_id::foo_id};
 
-    template <typename configuration_t>
-    constexpr auto invoke(configuration_t && cfg) const
-        requires detail::is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
-    {
-        return std::forward<configuration_t>(cfg).push_front(foo{2});
-    }
+    foo()                        = default;
+    foo(foo const &)             = default;
+    foo(foo &&)                  = default;
+    foo & operator=(foo const &) = default;
+    foo & operator=(foo &&)      = default;
+    ~foo()                       = default;
+
+    foo(t v) : value(v)
+    {}
+
+    t value;
 };
+//! [configuration_setup]
 
-struct with_bar_adaptor : public detail::configuration_fn_base<with_bar_adaptor>
+//! [compatibility]
+namespace seqan3::detail
+{
+template <>
+inline constexpr std::array<std::array<int, 2>, 2> compatibility_table<my_id>
 {
-    template <typename configuration_t>
-    constexpr auto invoke(configuration_t && cfg,
-                          float new_v) const
-        requires detail::is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
-    {
-        return std::forward<configuration_t>(cfg).push_front(bar{new_v});
-    }
-
-    template <typename configuration_t>
-    constexpr auto invoke(configuration_t && cfg) const
-        requires detail::is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
     {
-        return std::forward<configuration_t>(cfg).push_front(bar{2});
+        {0, 1},
```

Comment:
This looks much nicer than the code before :+1: 
I would add some more comments so the code is more self-explanatory 
```suggestion
        {0, 1}, // 'bar' cannot be combined with itself but with `foo`
```